### PR TITLE
PAYARA-4025 Setting context root with app.war:context stopped working in Payara Micro

### PR DESCRIPTION
**Pull request by MeroRai**

### CODE REVIEW

**Comment by MeroRai** [comment]
jenkins test please

**Comment by Pandrex247** [comment]
Jenkins test please

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java [hunk 1]
```diff
@@ -1518,7 +1516,7 @@ private void deployAll() throws GlassFishException {
                     File file = new File(entry);
                     String deployContext = file.getName();
                     String name = deployContext.substring(0, deployContext.length() - 4);
-                    if (deployContext.endsWith(".ear") || deployContext.endsWith(".war") || deployContext.endsWith(".jar") || deployContext.endsWith(".rar")) {
+                    if (hasJavaFileExtension(deployContext)) {
```
Comment (svendiedrichsen) [inline appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java]: Shouldn't this rather be called enterprise Java file extension?

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java [hunk 2]
```diff
@@ -1638,6 +1650,14 @@ private void deployAll() throws GlassFishException {
 
         LOGGER.log(Level.INFO, "Deployed {0} archive(s)", deploymentCount);
     }
+    
+    private boolean hasJavaFileExtension(String filePath) {
+        if (filePath.endsWith(".war") || filePath.endsWith(".ear") || filePath.endsWith(".jar") || filePath.endsWith(".rar")) {
```
Comment (svendiedrichsen) [inline appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java]: Simply return the if clause.

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java (hunk 1 above)
Comment (sgflt) [reply appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java]: Or hasJavaArchiveExtension.

**Comment by MeroRai** [comment]
jenkins test please

**Pull request merged**